apps/accounts/helpers.py

```diff
@@ -1,13 +1,48 @@
+import base64
+import json
+import os
+
 from django.conf import settings
 
+import jinja2
 from jingo import register
 
 
 @register.function
-def fxa_config():
-    return {camel_case(key): value
-            for key, value in settings.FXA_CONFIG.iteritems()
-            if key != 'client_secret'}
+@jinja2.contextfunction
+def fxa_config(context):
+    request = context['request']
+    config = {camel_case(key): value
+              for key, value in settings.FXA_CONFIG.iteritems()
+              if key != 'client_secret'}
+    set_fxa_state(request)
+    config['state'] = request.session['fxa_state']
+    return config
+
+
+def set_fxa_state(request):
+    if needs_fxa_state(request):
+        request.session['fxa_state'] = base64.urlsafe_b64encode(
```

Comment:
I had to base64 encode it because the spaces were being turned into `+`s when passed in the URL. I'm assuming that's something FxA does because I didn't see it being converted anywhere.